### ISSUE TITLE
Better time-out handling

### ISSUE THREAD

**Issue by zhengying-liu** [codalab/codalab-competitions#2577]
For competition: https://autodl.lri.fr/competitions/115

When a submission runs for a duration larger than the time budget (now set to 2 hours), one gets following error:
```
Traceback (most recent call last):
  File "/worker/worker.py", line 653, in run
    ingestion_process.kill()
  File "/usr/lib/python2.7/subprocess.py", line 1572, in kill
    self.send_signal(signal.SIGKILL)
  File "/usr/lib/python2.7/subprocess.py", line 1562, in send_signal
    os.kill(self.pid, sig)
OSError: [Errno 3] No such process
```
Besides, ingestion output and scoring output are empty.

But we'd like:
1. CodaLab will mark the submission as "Time-Out" instead of "Failed"
2. Better exception message saying "Time budget used up, killing the process..."
3. Produce scoring output and ingestion output
4. Still run scoring normally and produce a `scores.txt` file in output/ that can be used for parent scoring program

In summary, when a submission got time-out, CodaLab kills the ingestion process but will proceed normally for scoring.

Attached is a screen shot showing the error:
<img width="1274" alt="Screenshot 2019-04-14 at 09 42 16" src="https://user-images.githubusercontent.com/20074659/56090063-f8960080-5e9c-11e9-839b-7e1ce65f6087.png">

**Comment by zhengying-liu** [comment]
The same handing logic should be implemented when a participant clicks on "Cancel" button for a running submission made to Dataset 1, Dataset 2, ..., Dataset 5

**Comment by Tthomas63** [comment]
I believe Eric has fixed this, closing for now.
